lib/vagrant/util/downloader.rb

```diff
@@ -99,9 +99,28 @@ def download!
               # we report new progress reports. Otherwise, just keep
               # accumulating.
               match = progress_regexp.match(progress_data)
+
               break if !match
-              data = match[2]
-              progress_data.gsub!(match[1], "")
+
+              # If the download has been redirected and we are no longer downloading
+              # from the original host, notify the user that the target host has
+              # changed from the source.
+              if progress_data.include?("Location")
+                location = progress_data.scan(/Location: (.+?)$/m).flatten.compact.last.to_s.strip
+                if !location.empty?
+                  @logger.info("download redirected to #{location}")
+                  location_uri = URI.parse(location)
+                  source_uri = URI.parse(source)
+                  if location_uri.host != source_uri.host
+                    @ui.clear_line
+                    @ui.detail "Download redirected to host: #{location_uri.host}"
+                  end
+                end
+              end
+
+              data = match[1].to_s
+              stop = progress_data.index(data) + data.length
+              progress_data.slice!(0, stop)
```

Comment:
This seems to break at least when I'm trying to update boxes:

```
Checking for updates to 'bento/ubuntu-17.10'
Latest installed version: 201710.25.0
Version constraints: > 201710.25.0
Provider: virtualbox
Updating 'bento/ubuntu-17.10' with provider 'virtualbox' from version
'201710.25.0' to '201801.02.0'...
Loading metadata for box 'https://vagrantcloud.com/bento/ubuntu-17.10?access_token=cWHghvWylMrIug.atlasv1.DaEsXIBVKkcnY20HxjAivf26rvgjX939y8SAxzH1yCdeU4IYfUG6DU7pLyrzQHacwF8'
Adding box 'bento/ubuntu-17.10' (v201801.02.0) for provider: virtualbox
Downloading: https://vagrantcloud.com/bento/boxes/ubuntu-17.10/versions/201801.02.0/providers/virtualbox.box
Download redirected to host: archivist.vagrantup.com
Download redirected to host: vagrantcloud-files-production.s3.amazonaws.com
Progress: 100% (Rate: 335/s, Estimated time remaining: --:--:--)
```

It just sticks at that as line claiming 100%...where as prior to the change it showed the progress. Not sure if this is the case with regular box downloading too.